src/HealthGPS/analysis_module.cpp

```diff
@@ -149,11 +160,11 @@ AnalysisModule::calculate_residual_disability_weight(int age, const core::Gender
 void AnalysisModule::publish_result_message(RuntimeContext &context) const {
     auto sample_size = context.age_range().upper() + 1u;
     auto result = ModelResult{sample_size};
-    auto handle = core::run_async(&AnalysisModule::calculate_historical_statistics, this,
-                                  std::ref(context), std::ref(result));
+    // auto handle = core::run_async(&AnalysisModule::calculate_historical_statistics, this,
+    //                               std::ref(context), std::ref(result));
+    // handle.get();
```

Comment:
Is this not needed?

Reply:
Yeah good spot. It shouldn't be commented out. This is left over from when I was trying to debug the test failing. With these two lines uncommented, I get the following test failure;

```
        Start 236: TestAnalysisModule.CalculateIndex
236/236 Test #236: TestAnalysisModule.CalculateIndex ..............................***Failed    0.07 sec

Initialising with a custom GTest main function.

Using default test data store ...

Test location..: /home/ryan/projects/healthgps/out/build/linux-debug/src/HealthGPS.Tests
Test data store: /home/ryan/projects/healthgps/data

Note: Google Test filter = TestAnalysisModule.CalculateIndex
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from TestAnalysisModule
[ RUN      ] TestAnalysisModule.CalculateIndex
unknown file: Failure
C++ exception with description "map::at" thrown in the test fixture's constructor.

[  FAILED  ] TestAnalysisModule.CalculateIndex (57 ms)
[----------] 1 test from TestAnalysisModule (57 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (57 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestAnalysisModule.CalculateIndex

 1 FAILED TEST


99% tests passed, 1 tests failed out of 236

Total Test time (real) =  15.35 sec

The following tests FAILED:
        236 - TestAnalysisModule.CalculateIndex (Failed)
Errors while running CTest
```

I think this is because the `RuntimeContext` object that is created in the test fixture constructor is not properly initialised with the required map elements for some map that the commented-out `handle.get()` function above is trying to access.

Not quite there yet then. I feel like this is going to be a pain to sort out.

Reply:
Wasn't too bad in the end. The `RuntimeContext` needed the current year to be set.